aiida/orm/implementation/django/comment.py

```diff
@@ -7,126 +7,124 @@
-
-            # Constructing the default query
-            import operator
-            from django.db.models import Q
-            query_list = []
-
-            # If an id is specified then we add it to the query
-            if id is not None:
-                query_list.append(Q(pk=id))
-
-            # If a user is specified then we add it to the query
-            if user is not None:
-                query_list.append(Q(user=user))
-
-            # If a dbnode is specified then we add it to the query
-            if dbnode is not None:
-                query_list.append(Q(dbnode=dbnode))
-
-            res = DbComment.objects.filter(reduce(operator.and_, query_list))
-            ccount = len(res)
-            if ccount > 1:
-                raise MultipleObjectsError(
-                    "The arguments that you specified were too vague. More "
-                    "than one comments with this data were found in the "
-                    "database")
-            elif ccount == 0:
-                raise NotExistent("No comments were found with the given "
-                                 "arguments")
-
-            self.dbcomment = res[0]
+        try:
+            comment = DbComment.objects.get(id=comment_id)
+            return self.from_dbmodel(comment)
+        except ObjectDoesNotExist:
+            raise exceptions.NotExistent('Could not find Comment<{}>'.format(comment_id))
+        except MultipleObjectsReturned:
+            raise exceptions.ConfigurationError('Found multiple Comments for pk<{}>'.format(comment_id))
+
+    def delete(self, comment_id):
+        from django.core.exceptions import ObjectDoesNotExist
+        try:
+            DbComment.objects.get(pk=comment_id).delete()
+        except ObjectDoesNotExist:
+            raise exceptions.NotExistent("Comment with id '{}' not found".format(comment_id))
+
+    def from_dbmodel(self, dbmodel):
+        return DjangoComment.from_dbmodel(dbmodel, self.backend)
+
+
+class DjangoComment(entities.DjangoModelEntity[models.DbComment], BackendComment):
+    """Comment implementation for Django."""
+
+    MODEL_CLASS = models.DbComment
+
+    def __init__(self, backend, node, user, content=None):
+        """
+        Construct a DjangoComment.
+
+        :param node: a Node instance
+        :param user: a User instance
+        :param content: the comment content
+        :return: a Comment object associated to the given node and user
+        """
+        super(DjangoComment, self).__init__(backend)
+        type_check(user, users.DjangoUser)
+        self._dbmodel = utils.ModelWrapper(DbComment(node=node.dbnode, user=user.dbmodel, content=content))
 
     @property
     def pk(self):
-        return self.dbnode.pk
+        return self._dbmodel.pk
 
     @property
     def id(self):
-        return self.dbnode.pk
+        return self._dbmodel.pk
 
     @property
-    def to_be_stored(self):
-        return self.dbcomment.pk is None
+    def stored(self):
```

Comment:
`is_stored` is already in superclass.  Maybe redirect everything to that and delete this.